### PR TITLE
`z_obj_tokei_tobira` OK undocumented

### PR DESCRIPTION
**Pull request by AbdooOwd**
# `z_obj_tokei_tobira`
* OK (Matching)
* Undocumented

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:33]: You can remove the comment here

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:98]: You can rename this as `type`.
```suggestion
    s32 type = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
```
Don't forget to rename all the uses

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:122]: Leave an empty line between the variable declaration and the rest of the code
```suggestion
                    ObjTokeiTobira* tobira;

                    if (temp_param == OBJTOKEITOBIRA_TYPE_0) {
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:27]: We usually use this type of comment for the size of the structs
```suggestion
} ObjTokeiTobira; // size = 0x170
```

**Comment by AbdooOwd** [comment]
Only needs to run `./format.p`

**Review by AngheloAlf — APPROVED** [review]

**Comment by AngheloAlf** [comment]
Great work! PR looks good.

I'd like to mention that we are trying to get everybody who has contributed and/or wants to contribute to MM decomp to agree to licensing the repo before merging PRs from people who has not yet agreed to it. It is better explained here: https://github.com/louist103/mm-licensing/issues/1.

Could you please read about it and post in that issue if you agree or not. Thanks!

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:40]:
```suggestion
s16 D_80ABD76C[] = { -0x4000, 0x4000 };
```
0xC000 = -0x4000 as an `s16`

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:68]:
```suggestion
    if ((type == OBJTOKEITOBIRA_TYPE_0) && !CHECK_WEEKEVENTREG(WEEKEVENTREG_59_04) &&
```
Don't need the extra parens

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:155]:
```suggestion
    if ((this->unk164 * sp48) > 14336.0f) {
        this->unk160 *= 0.1f;
        this->unk164 = 14336.0f * sp48;
    } else if ((this->unk164 * sp48) < -1000.0f) {
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:146]:
```suggestion
        this->unk16C--;
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:115]:
```suggestion
                        this->unk16C = 80;
```
Looks like some kind of timer so is better as decimal

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:5]:
```suggestion
#define OBJTOKEITOBIRA_PARAMS(type) (type)
```
Kind of nitpicky, but plural seems better for these bit packing params macros.

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:69]:
```suggestion
        (play->sceneId == SCENE_CLOCKTOWER) && (gSaveContext.sceneLayer == 0) && (this->dyna.actor.csId > CS_ID_NONE)) {
```
Sorry missed this the first time.

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:43]:
```suggestion
s16 D_80ABD76C[] = {
    -0x4000, // OBJTOKEITOBIRA_TYPE_0
    0x4000,  // OBJTOKEITOBIRA_TYPE_1
};
CollisionHeader* D_80ABD770[] = {
    &object_tokei_tobira_Colheader_0012B0, // OBJTOKEITOBIRA_TYPE_0
    &object_tokei_tobira_Colheader_001590, // OBJTOKEITOBIRA_TYPE_1
};
f32 D_80ABD778[] = {
    1.0f,  // OBJTOKEITOBIRA_TYPE_0
    -1.0f, // OBJTOKEITOBIRA_TYPE_1
};
Gfx* D_80ABD780[] = {
    object_tokei_tobira_DL_001108, // OBJTOKEITOBIRA_TYPE_0
    object_tokei_tobira_DL_0013E8, // OBJTOKEITOBIRA_TYPE_1
};
```
All this data is indexed by type, so it's helpful to put comments on the data to visualize which type maps to which array element

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:66]:
```suggestion
    this->dyna.actor.world.pos.x += posOffset.x;
    this->dyna.actor.world.pos.y += posOffset.y;
    this->dyna.actor.world.pos.z += posOffset.z;
```
Position isn't really correct here. It's more an offset from the base position

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:96]:
```suggestion
    s32 pad1;
    ObjTokeiTobira* this = THIS;
    Player* player = GET_PLAYER(play);
    s32 pad2;
```
Usually if there's more than 1 pad, we name them `pad1` and `pad2`

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c:99]:
```suggestion
    s32 pad3;
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:7]:
```suggestion
#include "global.h"

#define OBJTOKEITOBIRA_GET_TYPE(thisx) ((thisx)->params & 1)
#define OBJTOKEITOBIRA_PARAMS(type) (type)
```
We typically have includes at the top of the file, so the params can be moved down

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h
Comment (engineer124) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:27]:
```suggestion
} ObjTokeiTobira; // size = 0x170 
```

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:5]:
```suggestion
#include "global.h"

#define OBJTOKEITOBIRA_GET_TYPE(thisx) ((thisx)->params & 1)
```
Small nit but can you add a newline here.

**Review by hensldm — APPROVED** [review]
LGTM. Btw, if someone has already approved the PR there is no need to request another review from them. I'm mentioning this since Anghelo had already approved your PR, but by requesting for a review it no longer shows up.

File: src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h
Comment (engineer124) [reply src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:5]: Bump

Comment (hensldm) [reply src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.h:5]: Would like to bump on this again and see if you could address this small little nit.

**Review by engineer124 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Comment by AbdooOwd** [comment]
LEZGOOO MERGED!!!11!!1!!!!

**Pull request merged**
